src/main/java/org/opensearch/knn/index/SpaceType.java

```diff
@@ -36,6 +36,14 @@ public float scoreTranslation(float rawScore) {
         public VectorSimilarityFunction getVectorSimilarityFunction() {
             return VectorSimilarityFunction.EUCLIDEAN;
         }
+
+        @Override
+        public float scoreToDistanceTranslation(float score) {
```

Comment:
are we adding the score to distance translation for L2 only? not for cosine and dot product?

Reply:
@navneet1v IMO no need for cosine and dot product here. The SpaceType Enum class is used for common translation for different engines:

* Cosine's score to distance function doesn't needed because Lucene's radial search api just need score parameter, in k-NN faiss, it doesn't support Cosine's type.
* Dot product's score to distance function is only need in Faiss engine and in Faiss it has different translation from Lucene, so I just put the function inside Lucene.java where it needed.